File: LarkCardKit/Models/Elements/Button.cs
using System.Text.Json.Serialization;
using LarkCardKit.Models.Behaviors;

namespace LarkCardKit.Models.Elements;

public class Button : Element
{
    public override string Tag => "button";

    [JsonPropertyName("type")]
    public string? Type { get; set; }
    
    [JsonPropertyName("size")]
    public string? Size { get; set; }
    
    [JsonPropertyName("width")]
    public string? Width { get; set; }
    
    [JsonPropertyName("text")]
    public PlainText? Text { get; set; }
    
    [JsonPropertyName("icon")]
    public object? Icon { get; set; }
    
    [JsonPropertyName("hover_tips")]
    public PlainText? HoverTips { get; set; }
    
    [JsonPropertyName("disabled")]
    public bool? Disabled { get; set; }
    
    [JsonPropertyName("disabled_tips")]
    public PlainText? DisabledTips { get; set; }
    
    [JsonPropertyName("confirm")]
    public ConfirmConfig? Confirm { get; set; }
    
    [JsonPropertyName("behaviors")]
    public List<object>? Behaviors { get; set; }
    
    [JsonPropertyName("name")]
    public string? Name { get; set; }
    
    [JsonPropertyName("form_action_type")]
    public string? FormActionType { get; set; }

    /// <summary>
    /// 点击行为（已废弃，请使用 behaviors 属性）
    /// </summary>
    [Obsolete("请使用 behaviors 属性代替")]
    [JsonPropertyName("onclick")]
    public object? OnClick { get; set; }
}

public class ConfirmConfig
{
    [JsonPropertyName("title")]
    public PlainText? Title { get; set; }
    
    [JsonPropertyName("text")]
    public PlainText? Text { get; set; }
}

File: Python/lark_card_kit/models/elements/button.py
"""Button element for interactive buttons."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any

from .base import Element
from .plain_text import PlainText


@dataclass
class ConfirmConfig:
    """Confirmation dialog configuration for buttons."""

    title: PlainText | None = None
    text: PlainText | None = None

    def to_dict(self) -> dict[str, Any]:
        """Convert to dictionary."""
        result: dict[str, Any] = {}
        if self.title:
            result["title"] = self.title.to_dict()
        if self.text:
            result["text"] = self.text.to_dict()
        return result


@dataclass
class Button(Element):
    """Button element for interactive buttons.

    Buttons can trigger actions like callbacks or URL opens.
    """

    tag: str = field(default="button", init=False)
    type: str | None = None
    size: str | None = None
    width: str | None = None
    text: PlainText | None = None
    icon: Any = None
    hover_tips: PlainText | None = None
    disabled: bool | None = None
    disabled_tips: PlainText | None = None
    confirm: ConfirmConfig | None = None
    behaviors: list[Any] | None = None
    name: str | None = None
    form_action_type: str | None = None
    onclick: Any = None